pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -630,6 +646,15 @@ func (r *ReconcileClusterDeployment) updateClusterDeploymentStatus(cd *hivev1.Cl
 		cdLog.Infof("cluster deployment status unchanged")
 	}
 
+	cdLog.Infof("REPORT DURATION METRIC FLAG:%v", reportDurationMetricsFlag)
+	// Report the cluster metrics if we need to
+	if reportDurationMetricsFlag {
+		cdLog.Infof("SETTING METRIC")
+		// jobDuration calculates the time elapsed since the install job started
+		jobDuration := job.Status.CompletionTime.Time.Sub(job.Status.StartTime.Time)
+		cdLog.WithField("duration", jobDuration.Seconds()).Debug("install job completed")
+		provisionJobsDurationHistogram.Observe(float64(jobDuration.Seconds()))
+	}
```

Comment:
I would move this block back into the original function right before "reconcile complete" just because there's no need to pass that boolean state around to a function that might get re-used elsewhere.

This looks fine however, it should work, I think you need some more testing, if the update fails it should immediately re-sync and try everything again.